pandas/tests/io/test_parquet.py

```diff
@@ -212,28 +212,42 @@ def check_error_on_write(self, df, engine, exc):
             with tm.ensure_clean() as path:
                 to_parquet(df, path, engine, compression=None)
 
-    def check_round_trip(self, df, engine, expected=None,
+    def check_round_trip(self, df, engine, expected=None, path=None,
                          write_kwargs=None, read_kwargs=None,
                          check_names=True):
+
         if write_kwargs is None:
-            write_kwargs = {}
+            write_kwargs = {'compression': None}
+
         if read_kwargs is None:
             read_kwargs = {}
-        with tm.ensure_clean() as path:
-            df.to_parquet(path, engine, **write_kwargs)
-            result = read_parquet(path, engine, **read_kwargs)
 
-            if expected is None:
-                expected = df
-            tm.assert_frame_equal(result, expected, check_names=check_names)
+        if expected is None:
+            expected = df
 
-            # repeat
-            to_parquet(df, path, engine, **write_kwargs)
-            result = pd.read_parquet(path, engine, **read_kwargs)
```

Comment:
well this is repeating quite a bit, just make an function def like this (make it a top-level function in this module, add a mini-doc string as well), and just call it.

```
def check_round_trip(engine, path, df, expected, write_kwargs, read_kwargs, check_names):
    df.to_parquet(path, engine, **write_kwargs)
    result = read_parquet(path, engine, **read_kwargs)
    tm.assert_frame_equal(expected, actual,
                                      check_names=check_names)
```

Reply:
I think that your suggestion is similar to the idea the was rejected of [`do_round_trip`](https://github.com/pandas-dev/pandas/pull/19135/commits/230c8148dad1739873489e52d663f8e376bdb814#diff-539eb41a29c68cc27818db4b480c2fd6R215) method.

Maybe a better name other then `do_round_trip` would be `_round_trip` inside `Base` class?

Reply:
pls do it my way

Reply:
this needs updating